pkg/controllers/clusterresourceplacementeviction/controller.go

```diff
@@ -0,0 +1,257 @@
+		}
+		markEvictionExecuted(&eviction, evictionAllowedNoPDB)
+		return runtime.Result{}, r.updateEvictionStatus(ctx, &eviction)
+	}
+
+	var desiredBindings int
+	switch crp.Spec.Policy.PlacementType {
+	case placementv1beta1.PickAllPlacementType:
+		desiredBindings = len(crbList.Items)
+	case placementv1beta1.PickNPlacementType:
+		desiredBindings = int(*crp.Spec.Policy.NumberOfClusters)
+	case placementv1beta1.PickFixedPlacementType:
+		desiredBindings = len(crp.Spec.Policy.ClusterNames)
+	}
+
+	totalBindings := len(crbList.Items)
+	allowed, disruptionsAllowed, availableBindings := isEvictionAllowed(desiredBindings, crbList.Items, db)
+	if allowed {
+		if err := r.deleteClusterResourceBinding(ctx, evictionTargetBinding); err != nil {
+			return runtime.Result{}, err
+		}
+		markEvictionExecuted(&eviction, fmt.Sprintf(evictionAllowedPDBSpecified, disruptionsAllowed, availableBindings, desiredBindings, totalBindings))
+	} else {
+		markEvictionNotExecuted(&eviction, fmt.Sprintf(evictionBlockedPDBSpecified, disruptionsAllowed, availableBindings, desiredBindings, totalBindings))
+	}
+
+	return runtime.Result{}, r.updateEvictionStatus(ctx, &eviction)
+}
+
+// updateEvictionStatus updates eviction status.
+func (r *Reconciler) updateEvictionStatus(ctx context.Context, eviction *placementv1alpha1.ClusterResourcePlacementEviction) error {
+	evictionRef := klog.KObj(eviction)
+	if err := r.Client.Status().Update(ctx, eviction); err != nil {
+		klog.ErrorS(err, "Failed to update eviction status", "clusterResourcePlacementEviction", evictionRef)
+		return controller.NewUpdateIgnoreConflictError(err)
+	}
+	klog.V(2).InfoS("Updated the status of a eviction", "clusterResourcePlacementEviction", evictionRef)
+	return nil
+}
+
+// deleteClusterResourceBinding deletes the specified cluster resource binding.
+func (r *Reconciler) deleteClusterResourceBinding(ctx context.Context, binding *placementv1beta1.ClusterResourceBinding) error {
+	bindingRef := klog.KObj(binding)
+	if err := r.Client.Delete(ctx, binding); err != nil {
+		klog.ErrorS(err, "Failed to delete cluster resource binding", "clusterResourceBinding", bindingRef)
+		return controller.NewDeleteIgnoreNotFoundError(err)
+	}
+	klog.V(2).InfoS("Issued delete on cluster resource binding, eviction succeeded", "clusterResourceBinding", bindingRef)
+	return nil
+}
+
+// isEvictionAllowed calculates if eviction allowed based on available bindings and spec specified in placement disruption budget.
+func isEvictionAllowed(desiredBindings int, bindings []placementv1beta1.ClusterResourceBinding, db placementv1alpha1.ClusterResourcePlacementDisruptionBudget) (bool, int, int) {
+	availableBindings := 0
+	for i := range bindings {
+		availableCondition := bindings[i].GetCondition(string(placementv1beta1.ResourceBindingAvailable))
+		if condition.IsConditionStatusTrue(availableCondition, bindings[i].GetGeneration()) {
+			availableBindings++
+		}
+	}
+	var disruptionsAllowed int
+	if db.Spec.MaxUnavailable != nil {
```

Comment:
We have API validation to prevent this concern https://github.com/Azure/fleet/blob/main/apis/placement/v1alpha1/disruptionbudget_types.go#L30

But I do agree that there should only be one outcome though